src/coreclr/inc/corinfo.h

```diff
@@ -1738,6 +1739,20 @@ struct CORINFO_THREAD_STATIC_BLOCKS_INFO
     uint32_t offsetOfGCDataPointer;
 };
 
+//----------------------------------------------------------------------------
+// getThreadLocalStaticInfo_ReadyToRun and CORINFO_THREAD_STATIC_INFO_READYTORUN: The EE instructs the JIT about how to access a thread local field
+
+struct CORINFO_THREAD_STATIC_INFO_READYTORUN
+{
+    uint32_t offsetOfThreadLocalStoragePointer;
+    CORINFO_CONST_LOOKUP tlsRootObject;
+    CORINFO_CONST_LOOKUP tlsIndexObject;
+    CORINFO_CONST_LOOKUP threadStaticBaseSlow;
+    uint32_t classCtorContextSize;
```

Comment:
This duplicates information returned by `getIsClassInitedFlagAddress`. (This is related to my other comment about static constructors.)

I think it would be better to call `getIsClassInitedFlagAddress` and friends, just like we do for regular static constructors expansion instead of duplicating the logic here.

Reply:
I was thinking of reusing `getIsClassInitedFlagAddress`, but I would have to anyway call some method to get the TLS info like `tlsRootObject`, etc., so I will just have a dedicated method for it, which will return everything needed for TLS.